packages/signers/src/message-partial-signer.ts

```diff
@@ -0,0 +1,29 @@
+import { Address } from '@solana/addresses';
+
+import { SignableMessage } from './signable-message';
+import { SignatureDictionary } from './types';
+
+/** Defines a signer capable of signing messages. */
+export type MessagePartialSigner<TAddress extends string = string> = Readonly<{
+    address: Address<TAddress>;
+    signMessage(messages: readonly SignableMessage[]): Promise<readonly SignatureDictionary[]>;
```

Comment:
nit: This was a little confusing at first, since it's called `signMessage` but it actually can sign multiple.

Reply:
Glad you mentioned it. It's also been on my mind.

This is the current API:

```ts
const [signatures] = await signer.signMessage([message]);
const [signaturesA, signaturesB] = await signer.signMessage([messageA, messageB]);
```

These are the alternatives:

```ts
// Option A: Use a variadic argument (this is what the wallet-standard does).
const [signatures] = await signer.signMessage(message);
const [signaturesA, signaturesB] = await signer.signMessage(messageA, messageB);

// Option B: Pluralize.
const [signatures] = await signer.signMessages([message]);
const [signaturesA, signaturesB] = await signer.signMessages([messageA, messageB]);
```

IMO, they each have pros and cons:
- **Current API**: most of the time we're only signing one message or transaction so `signMessage` reads better but there's confusion with the fact that we provide an array.
- **Option A**: Uses `signMessage` + no longer provides an array + consistent with the wallet standard. However, we loose the type symmetry between input and output. It feels a bit weird that you pass an object but get an array of objects.
- **Option B**: Most predictable and consistent API but we're having to use `signMessages` even when most of the time we want to sign single messages.

That being said, I prefer Option A or B over the current API. What about you?

Reply:
I worry that it won't be obvious enough to everyone that option A can be used to sign multiple messages in parallel if only you supply more than one message as an argument, so I prefer option B.

Reply:
Yeah I’m also on team B!

Reply:
✅ Updated the API using Option B.